feat: List recipes at main

Lahavas opened the pull request:
- Add pagination logic in controller and service layer
- Add recipe list acceptance test and service test
- Create FixtureFactory to generate recipe list

Lahavas committed:
feat: Create recipe's page backend logic

- Add pagination logic in controller and service layer
- Add recipe list acceptance test and service test
- Create FixtureFactory to generate recipe list

Lahavas commented:
아직 Frontend 부분은 완료가 안됐기 때문에 아직 Merge는 말아주세요 :)

ryuneeee commented on src/test/java/com/woowahan/techcamp/recipehub/recipe/controller/RecipeListAcceptanceTest.java:
```diff
@@ -0,0 +1,114 @@
+
+    private static final String REQUEST_PARAM_PAGE_NUMBER = "page";
+    private static final String REQUEST_PARAM_PAGE_SIZE = "size";
+    private static final String REQUEST_PARAM_PAGE_SORT = "sort";
```
흠.. 빼는건 좋은거 같은데, 다르게 보면 또 23자리 읽는것보단 3자리 읽는게 더 명확한거 같기도 하고요 흠흠...
뭐 좋은방법 없을까요?

ryuneeee commented on src/test/java/com/woowahan/techcamp/recipehub/recipe/service/RecipeListServiceTest.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    @Test
+    public void getCorrectPageThatSortedAscendingOrderByNameProperty() {
```
CorrectPage는 어떤걸 의미해요? 그냥 정상적인 페이지 출력이라면 줄이는게 나을것 같아요.
어차피 우리는 정상적으로 되는게 테스트를 하는 목적이니까요. 

이게 나름, 변수/함수명 짧게 짓는것도 전 매우 중요하다고 생각해요.
생각보다 CamelCase가 길어지면 가독성이 썩 안좋은거 같다는...

Lahavas replied:
이게 정렬 방향이랑 Property가 변경되는 케이스가 있어서 피치못하게... 대안책이 있을때 수정하여 반영하겠습니다 👍

ryuneeee commented on src/test/java/com/woowahan/techcamp/recipehub/support/util/FixtureFactory.java:
```diff
@@ -0,0 +1,26 @@
+import java.util.Date;
+import java.util.List;
+
+public class FixtureFactory {
```
이런거 좋좋 👍

ryuneeee commented on src/test/java/com/woowahan/techcamp/recipehub/recipe/service/RecipeListServiceTest.java:
```diff
@@ -0,0 +1,73 @@
+    public void createPageInfo() throws Exception {
+        lastRecipeList = generateRecipeList(5);
+
+        sort = new Sort(Sort.Direction.ASC, SORT_PROPERTY);
```
이런건 static 보다는 그냥 "id"로 써주는게 코드를 읽어나갈때 더 명확한거 같아요. 
그래서 SORT_PROPERTY가 뭐지? 하고 다시 올려보게 되거든요. 

```assertThat(resultRecipePage.getNumber()).isEqualTo(PAGE_NUMBER);```
바로 밑에 존재하는 테스트 코드 같은 경우에는, getNumber()를 호출해서 비교하는것이기 때문에
사실 EqualTo쪽에는 Static 변수보다는 그냥 숫자가 나을수도 있을꺼 같구요.

너무 과하게만 사용하지 말아주세요👍

Lahavas committed:
Merge branch 'develop' into feature/paging-recipe-list

Lahavas committed:
Merge branch 'develop' into feature/paging-recipe-list

# Conflicts:
#	src/main/java/com/woowahan/techcamp/recipehub/recipe/controller/RecipeController.java
#	src/main/java/com/woowahan/techcamp/recipehub/recipe/domain/Recipe.java

ryuneeee commented:
지금 보니 왜이리 리뷰에 날이 서있죠 -.,-...ㅋ 코드리뷰가 꼭 안좋은 코드에 대해서만 리뷰하는게 아닐텐데... 다음에 조금 더 신경 쓸게요ㅎㅎ

Lahavas committed:
Merge branch 'develop' into feature/paging-recipe-list

# Conflicts:
#	src/main/java/com/woowahan/techcamp/recipehub/recipe/controller/RecipeController.java
#	src/main/java/com/woowahan/techcamp/recipehub/recipe/domain/Recipe.java

Lahavas committed:
feat: Add PageListDTO to show pagination list

- Fix acceptance test to pagination list
- Add unit test for PageListDTO and PageItemDTO
- Add zeroBased Method in HomeController to convert one-based

Lahavas committed:
feat: Add pagination list in index.hbs (F/E)

Lahavas commented:
Resolves: #49

Lahavas committed:
Merge branch 'develop' into feature/paging-recipe-list
